### ISSUE TITLE
How to dump correct memory in game guardian to get libil2cpp.so file?

### ISSUE THREAD

**Issue by teamjrmods** [Perfare/Il2CppDumper#257]
- I have tried dumping with Il2CppDumper-v6.1.2 but it shows the following error

Initializing metadata...
Initializing il2cpp file...
Applying relocations...
WARNING: find .init_proc
**ERROR: This file is protected.**

- I know the file is protected so I did a memory dump in Game Guardian

- The problem is that I get .bin extension files and I don't know how to find the .so extension file to run it in Il2CppDumper-v6.1.2

Please help I don't know what I'm doing wrong .. !!
**Sorry for my English**

**Comment by Perfare** [comment]
Check the `*-maps.txt` file in the GameGuardian dump folder, determine the address of the `libil2cpp.so`, find the bin file containing the address, if the file name is the same as the `libil2cpp.so` address, just load directly, otherwise trim the beginning of the address difference.

**Comment by teamjrmods** [comment]
Thanks, I already solved the problem..

**Comment by sengc92** [comment]
Hello
I have encountered difficulties while trimming the file to libil2cpp.so, ELF was not located at the correspond address , hope to get some insights here.

using game guardian:
997ae000-9ee4c000 rw-p 00000000 08:13 819273     /data/app/--/lib/arm/libil2cpp.so
9ee4c000-9ee4d000 r--p 0569e000 08:13 819273     /data/app/--/lib/arm/libil2cpp.so
9ee4d000-9ee4e000 r--p 0569f000 08:13 819273     /data/app/--/lib/arm/libil2cpp.so
9ee4e000-9ee50000 rw-p 056a0000 08:13 819273     /data/app/--/lib/arm/libil2cpp.so
 
*****-8e7b5000-a12d2000.bin   306292KB

the correspond address should be:
8e7b5000 - 997ae000 = AFF9000
9ee50000 - 8e7b5000 = 1069B000

I ran multiple instances and get the similar results as shown in pictures. 
Should I just trim all ELFs available inside the file and test with II2CppDumper?
Or they have been shifted to some other order?.. That will be beyond my understanding.

![image](https://user-images.githubusercontent.com/63807395/79486410-82eed400-7fcb-11ea-8930-d2efe86a728d.png)
![image](https://user-images.githubusercontent.com/63807395/79486746-014b7600-7fcc-11ea-8558-7fb865e793e4.png)


Thanks, 
I will be doing more testing later.
